### ISSUE TITLE
Update Available Message but there is no update

### ISSUE THREAD

**Issue by tartas1995** [pop-os/shop#198]
**Distribution (run `cat /etc/os-release`):**
 NAME="Pop!_OS"
VERSION="20.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.04 LTS"
VERSION_ID="20.04"
HOME_URL="https://system76.com/pop"
SUPPORT_URL="http://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
LOGO=distributor-logo-pop-os

**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
pop-shop:
  Installed: 3.3.0pop0~1589405854~20.04~72b0ae9
  Candidate: 3.3.0pop0~1589405854~20.04~72b0ae9
  Version table:
 *** 3.3.0pop0~1589405854~20.04~72b0ae9 1001
       1001 http://ppa.launchpad.net/system76/pop/ubuntu focal/main amd64 Packages
        100 /var/lib/dpkg/status

**Issue/Bug Description:**
I have 2 users. (1 sudo/1 normal). on the normal user, I get the Message "Update Available" and a "1" badge in the shop on the installed tab but there is no "update all" button or any "update" button. apt update and apt upgrade doesn't fix it too. on the sudo user, I don't get any Message or badge. I am fairly sure that there is no update.

**Expected behavior:**
no Message "Update Available" and not badge on the tab if there is no Update Available.
or
let me update.

**Other Notes:**
Just very annoying and confusing.

**Comment by olivier17430** [comment]
Hello,
maybe a flatpak update ?
did you try the command
`$ flatpak update`
in a terminal ?

**Comment by dephekt** [comment]
> Hello,
> maybe a flatpak update ?
> did you try the command
> `$ flatpak update`
> in a terminal ?

I had the same issue as the reporter. Running `flatpak update` showed an update for openh264. I ran the update and reopened Pop Shop and now the badge is gone. Thanks for the suggestion!

**Comment by tartas1995** [comment]
Yesterday, I tried restarting the machine and it stayed but today it was gone. I don't know why it was there or why is disappeared. I can't provide any more information. I could close it but I think it is a bug and that It should stay open by of it.

**Comment by Philbohackinz** [comment]
For me flatpak update works but I've suggested to others to try and it wasn't working for them.   I wonder if there is anything else that could cause this issue.

**Comment by tartas1995** [comment]
OK, it came back and now i am running 
`$ flatpak update`
updated stuff but there message is still appearing.

**Comment by AllizomFoxfire** [comment]
> flatpak update

Great, it fixed the issue. However the larger and still unresolved issue is that Pop!_Shop identifies the update yet can't apply it.

**Comment by TC68** [comment]
NAME="Pop!_OS"
VERSION="20.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.04 LTS"
VERSION_ID="20.04"
HOME_URL="https://system76.com/pop"
SUPPORT_URL="http://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
LOGO=distributor-logo-pop-os


*$ flatpak update* seems to fixed it for me

Offending item was:
 1. [✓] org.freedesktop.Platform.openh264            2.0              u            flathub           1.2 MB / 1.5 MB

**Comment by cassidyjames** [comment]
I believe this is an issue that we've addressed upstream: https://github.com/elementary/appcenter/pull/1359

**Comment by MikeyBeez** [comment]
It seems that parts of the OS are flatpaks, but they don't show up as applications, because they aren't user apps. If you hit the settings wheel, then hit info, all the flatpaks are listed.   Or you can type flatpak list in the terminal and see them in a better format.  BTW, it was driving my OCD self nuts too. ;)   Thanks for the great OS System 76.  I tell everyone it's the best.

**Comment by Oymate** [comment]
![Screenshot from 2020-06-28 13-23-56](https://user-images.githubusercontent.com/50857856/85941115-babab600-b942-11ea-8d6a-f6af10f263ba.png)
Screenshot for reference

**Comment by MikeyBeez** [comment]
Go to a terminal and run sudo flatpak update.

**Comment by AllizomFoxfire** [comment]
Agreed. Whilst the solution may be that simple, if the OS contains items
that have applications installed from several repositories and installation
methods then the Pop Shop should also be capable of updating those items
without the user having to take extraneous steps as updates become
available.

Thanks,

Shawn Egan

**Comment by piaccarino** [comment]
Since this one is open, I'll add here but can open a new issue if it makes sense:
NAME="Pop!_OS"
VERSION="20.10"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.10"
VERSION_ID="20.10"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=groovy
UBUNTU_CODENAME=groovy
LOGO=distributor-logo-pop-os

Have the installed 1 bubble with nothing to update in the shop.

flatpak update -v
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /root/.local/share/flatpak
Looking for updates…
Nothing to do.

sudo flatpak repair -v
F: Opening system flatpak installation at path /var/lib/flatpak
Working on the system installation at /var/lib/flatpak
Pruning objects
F: Pruning repo
F: Pruned 0/0 objects, size 0 bytes
F: No installations directory in /etc/flatpak/installations.d. Skipping

sudo flatpak uninstall --unused
Nothing unused to uninstall

sudo snap refresh
All snaps up to date.

sudo apt full-upgrade
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

I literally cannot find what it thinks needs updating and I would love to know where it is finding that since there is no bubble until after it does its check (the app icon has the bubble all the time though).  Started around a month ago and figured it would clear out at some point.

**Comment by Oymate** [comment]
flatpak repair --user
flatpak update

**Comment by piaccarino** [comment]
ID                          Branch         Op         Remote          Download
 1. [✓] org.gnome.Platform          3.36           u          flathub         860.4 kB / 326.0 MB

Thanks @Oymate, that was it!

**Comment by Metaph413** [comment]
> flatpak repair --user
> flatpak update

This worked for me too.

**Comment by bassmanitram** [comment]
> flatpak repair --user
> flatpak update

FINALLY! Thanks

**Comment by Oymate** [comment]
Short-term solution could be running the repair command with a button.

**Comment by QEDeD** [comment]
I've several times experienced that the store would show updates being available, but there wouldn't be any available to update in the store. As far as I can tell, this has always happened after I (tried to) installed all available updates.

When I run `flatpak update` it becomes that the "invisible" updates are in fact flatpak updates that I can update manually via the CLI.

Before manual update:

![before manual update](https://user-images.githubusercontent.com/16562457/108604168-60c4a000-73ac-11eb-90dc-e715adc5098e.png)

After manual update:

![image](https://user-images.githubusercontent.com/16562457/108604179-75a13380-73ac-11eb-8457-5fc6f16773ab.png)

**Comment by APCBoston** [comment]
I had this issue as well, and none of the low-hanging suggestions (`flatpak repair --user`, `flatpak update appstream`, `flatpak install --unused`) helped at all.  It was pretty clearly a `flatpak` issue, and the solution was ultimately [here](https://github.com/flatpak/flatpak/issues/4618) -- run `ostree fsck --repo=.local/share/flatpak/repo` to clean out the corrupted flatpak objects.  `ostree` isn't installed by default on Pop!, so this also requires `sudo apt install ostree`.  

If System76 wanted to "fix" this issue in the Pop! Shop, it would probably need to involve detecting the error and running the fsck automatically to clean the repo.

**Comment by Abyss-L** [comment]
I have the same issue and none of the suggestions helped (not even ostree). Fresh installed Pop!_OS 22.04 a week ago.
